dash/_configs.py

```diff
@@ -0,0 +1,104 @@
+    if init is not None:
+        return init
+
+    env_value = env.get('DASH_{}'.format(config_name.upper()))
+    if env_value is None:
+        return default
+    return env_value
+
+
+def pathname_configs(url_base_pathname=None,
+                     routes_pathname_prefix=None,
+                     requests_pathname_prefix=None,
+                     environ_configs=None):
+    _pathname_config_error_message = '''
+    {} This is ambiguous.
+    To fix this, set `routes_pathname_prefix` instead of `url_base_pathname`.
+
+    Note that `requests_pathname_prefix` is the prefix for the AJAX calls that
+    originate from the client (the web browser) and `routes_pathname_prefix` is
+    the prefix for the API routes on the backend (this flask server).
+    `url_base_pathname` will set `requests_pathname_prefix` and
+    `routes_pathname_prefix` to the same value.
+    If you need these to be different values then you should set
+    `requests_pathname_prefix` and `routes_pathname_prefix`,
+    not `url_base_pathname`.
+    '''
+    environ_configs = environ_configs or env_configs()
+
+    url_base_pathname = choose_config('url_base_pathname',
+                                      url_base_pathname,
+                                      environ_configs)
+
+    routes_pathname_prefix = choose_config('routes_pathname_prefix',
+                                           routes_pathname_prefix,
+                                           environ_configs)
+
+    requests_pathname_prefix = choose_config('requests_pathname_prefix',
+                                             requests_pathname_prefix,
+                                             environ_configs)
+
+    if url_base_pathname is not None and requests_pathname_prefix is not None:
+        raise exceptions.InvalidConfig(
+            _pathname_config_error_message.format(
+                'You supplied `url_base_pathname` and '
+                '`requests_pathname_prefix`.'
+            )
+        )
+    elif url_base_pathname is not None and routes_pathname_prefix is not None:
+        raise exceptions.InvalidConfig(
+            _pathname_config_error_message.format(
+                'You supplied `url_base_pathname` and '
+                '`routes_pathname_prefix`.')
+        )
+    elif url_base_pathname is not None and routes_pathname_prefix is None:
+        routes_pathname_prefix = url_base_pathname
+    elif routes_pathname_prefix is None:
+        routes_pathname_prefix = '/'
+
+    if not routes_pathname_prefix.startswith('/'):
+        raise exceptions.InvalidConfig(
+            '`routes_pathname_prefix` needs to start with `/`')
+    if not routes_pathname_prefix.endswith('/'):
+        raise exceptions.InvalidConfig(
+            '`routes_pathname_prefix` needs to end with `/`')
+
+    app_name = environ_configs.DASH_APP_NAME
+
+    if not requests_pathname_prefix and app_name:
+        requests_pathname_prefix = '/' + app_name + routes_pathname_prefix
```

Comment:
It took me a while to understand this but I think I finally get it. This is the case for users that are deploying on DDS that want a different URL base name. i.e. instead of just `/my-app-name` they want it to be `/some-base-url/another-app`. I think this will be uncommon but it makes sense and it seems complete 👍